jerry-ext/args/args-internal.h

```diff
@@ -0,0 +1,31 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_ARGS_INTERNAL_H
+#define JERRYX_ARGS_INTERNAL_H
+
+#include "jerryscript.h"
+
+/**
+ * The iterator structor for JS arguments.
+ */
+struct jerryx_arg_js_iterator_t
```

Comment:
Another place for declaring the struct would be arg-js-iterator-helper.c and not even having an internal header. This  would ensure that even the other components of this library would be blocked from accessing the internals of the data structure and would be forced to use the accessor methods. I'm not insisting but that would be perhaps the "nicest" or "cleanest" approach.